.github/workflows/python-gh-release.yml

```diff
@@ -0,0 +1,50 @@
+name: Does GitHub release for given version and extracts release notes from CHANGELOG.md
+
+on:
+  workflow_call:
+    inputs:
+      VERSION:
+        required: true
+        type: string
+        description: The version being released
+
+jobs:
+  do-release:
+    name: Do Release
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Extract Release Notes
+      run: |
+        RELEASE="${{ inputs.VERSION }}"
+        STARTED=0
+        while IFS= read -r LINE; do
+          if [ ${STARTED} -eq 1 ]; then
+            if [[ "${LINE}" == \##\ * ]]; then
+              break
+            fi
+            echo "${LINE}" >> ${{ github.workspace }}-release-notes.txt
+          elif [[ "${LINE}" == \#\#\ [\[?]*${RELEASE}[\]?]* ]]; then
+            STARTED=1
+            echo "${LINE}" >> ${{ github.workspace }}-release-notes.txt
+          fi
+        done < "CHANGELOG.md"
+        
+        if [ ${STARTED} -eq 0 ]; then
+          echo "Release ${RELEASE} not found in CHANGELOG.md, no CHANGELOG release notes available" 1>&2
+        else
+          echo "Detected Release Notes for ${RELEASE} are:"
+          cat ${{ github.workspace }}-release-notes.txt
+        fi
+        exit 0
```

Comment:
assumption: Unable can't use release-please's release note generation, because something in your python publishing generates `CHANGELOG.md`

Reply:
We do use release please to generate the notes and create a feature branch, but we're using this workflow to do the final release itself.

That's mostly because release please is working in a different workflow, so there's a non-zero chance that our own workflow would get to the point of attaching sboms to the release before the release was created. This approach is how the Java team do their releases, so for now I think it's a simple way to handle it.

In the future we might look to go back to release please for RCs or such.